Add version info to development builds loading panel

csmaxpower opened the pull request:
When the `DEVELOPMENT BUILD` label was added in (#2969), `ETLEGACY_VERSION` was removed due to space constraints.  The version information is extremely useful to have so this commit aims to solve both the need for the development build label as well as having version information on the loading panel.

Positioning of the start of the print was adjusted to account for having two lines versus one, as well as text sizing adjustments to ensure two lines are still readable within the desired area.  Keeping the version on a separate line also ensures space for longer build strings should it be necessary over the current example below rendered by this commit:


![image](https://github.com/user-attachments/assets/607054a2-6e1c-4efc-b332-60760ab91404)

csmaxpower committed:
Add version info to development builds loading panel

When the `DEVELOPMENT BUILD` label was added, `ETLEGACY_VERSION` was removed due to space constraints.  The version information is extremely useful to have so this commit aims to solve both the need for the development build label as well as having version information on the loading panel.

Positioning of the start of the print was adjusted to account for having two lines versus one, as well as text sizing adjustments to ensure two lines are still readable within the desired area.  Keeping the version on a separate lines also ensures space for longer build strings should they be necessary over the current example.

Vorschreibung commented:
@csmaxpower 

1. Upper-text sticks a lot towards the top border - can you possibly move it a tad down?
2. Distance between Upper-text and Lower-text seems a bit much - can you possibly reduce it?
3. Lower-text draws onto the border of the text box - can you possibly move it up?
4. Was it on purpose that you switched `DEVELOPMENT BUILD` & `legacy` ? - This makes it now inconsistent to the normal version display that's shown when you don't play on a development build, see second attachment. - Similarly the ingame menu also states:
```
ETLEGACY
DEVELOPMENT BUILD
```
and not the other way around.

![image](https://github.com/user-attachments/assets/83d74c27-2ca6-486f-90e7-62743ed9a8df)

### Release build
![image](https://github.com/user-attachments/assets/5073cdb6-3fbb-4b8f-a316-3f09eec32b77)

Vorschreibung commented on src/cgame/cg_loadpanel.c:
```diff
@@ -341,10 +341,12 @@ void CG_DrawConnectScreen(qboolean interactive, qboolean forcerefresh)
 		}
 		else
-		{
+		{			
```
@csmaxpower trailing spaces - please make sure you have the https://editorconfig.org/ plugin for your Editor/IDE enabled and push again

Vorschreibung commented on src/cgame/cg_loadpanel.c:
```diff
@@ -341,10 +341,12 @@ void CG_DrawConnectScreen(qboolean interactive, qboolean forcerefresh)
 		{
-			CG_Text_Paint_Centred_Ext(x, y, 0.22f, 0.22f, clr3, va("^1%s ^8DEVELOPMENT BUILD", MODNAME), 0, 0, 0, &cgs.media.bg_loadscreenfont1);
+			y = 317;
+			CG_Text_Paint_Centred_Ext(x, y, 0.20f, 0.20f, clr3, va("^8DEVELOPMENT BUILD ^1%s", MODNAME), 0, 0, 0, &cgs.media.bg_loadscreenfont1);
```
@csmaxpower Like mentioned in https://github.com/etlegacy/etlegacy/pull/2998#issuecomment-2502158246 - are you sure you want to swap the va string parts? - If so, shouldn't you also switch them for line 346?

csmaxpower committed:
Adjust text positioning in original PR

> Upper-text sticks a lot towards the top border - can you possibly move it a tad down?

Moved top line text lower (text size decrease was the better option after playing with a variety of padding options) away from top border.

> Distance between Upper-text and Lower-text seems a bit much - can you possibly reduce it?

Reduced padding between upper and lower text lines by raising the start position of the bottom line.

> Lower-text draws onto the border of the text box - can you possibly move it up?

Bottom line text has been moved up to not spill into border graphic below.

> Was it on purpose that you switched DEVELOPMENT BUILD & legacy ? - This makes it now inconsistent to the normal version display that's shown when you don't play on a development build, see second attachment. - Similarly the ingame menu also states:

It was not on purpose, was left by mistake when originally tinkering.  It has been reverted back to original form.

csmaxpower commented:
> * Upper-text sticks a lot towards the top border - can you possibly move it a tad down?
> * Distance between Upper-text and Lower-text seems a bit much - can you possibly reduce it?
> * Lower-text draws onto the border of the text box - can you possibly move it up?
> * Was it on purpose that you switched `DEVELOPMENT BUILD` & `legacy` ? - This makes it now inconsistent to the normal version display that's shown when you don't play on a development build, see second attachment. - Similarly the ingame menu also states:

New commit with the above requested changes has been made, including the reverting back to original label text sequence of `MODNAME` & `DEVELOPMENT BUILD`, which was left by mistake when trying to fit it all on one line originally.  See commit notes for further details on adjustments.

![image](https://github.com/user-attachments/assets/fb73ca37-8309-46d6-b789-ad5894ae43b1)

Vorschreibung commented:
@csmaxpower LGTM, thanks for the patch!

Aranud commented:
Thank you !
